rosbag2_cpp/src/rosbag2_cpp/clocks/time_controller_clock.cpp

```diff
@@ -114,6 +115,72 @@ class TimeControllerClockImpl
     snapshot(ros_now());
   }
 
+  /**
+   * \brief Adjust internal clock to the specified timestamp.
+   * \details It will change the current internally maintained offset so that next published time
+   * is different.
+   * \note Will trigger any registered JumpHandler callbacks.
+   * \param ros_time Time point in ROS playback timeline.
+   */
+  void jump(rcutils_time_point_value_t ros_time)
+  {
+    rcl_duration_t time_jump_delta;
+    {
+      std::lock_guard<std::mutex> lock(state_mutex);
```

Comment:
Since ROS 2 supports C++ 17, we now have access to the `std::scoped_lock` class. Should we use that here?

Reply:
We haven't yet changed rosbag2 to target C++17 - this is an upgrade that could be done in a new PR if we're inclined